cmd/dockerd/daemon.go

```diff
@@ -83,6 +83,13 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 	if cli.Config, err = loadDaemonCliConfig(opts); err != nil {
 		return err
 	}
+
+	if opts.Validate {
+		// If config wasn't OK we wouldn't have made it this far.
+		logrus.Infof("Config OK")
```

Comment:
```bash
mkdir /foo
echo '{' > /foo/malformed-config.json
echo '' > /foo/empty-config-1.json
echo '{}' > /foo/empty-config-2.json
echo '{"unknown-option": true}' > /foo/invalid-config-1.json
echo '{"debug": true}' > /foo/valid-config-1.json
```

Then trying them;

```bash
for file in /foo/*.json; do sh -xc "dockerd --validate --config-file=${file}"; done
+ dockerd --validate --config-file=/foo/empty-config-1.json
unable to configure the Docker daemon with file /foo/empty-config-1.json: EOF
+ dockerd --validate --config-file=/foo/empty-config-2.json
INFO[2021-05-19T10:31:30.747037342Z] Config OK
+ dockerd --validate --config-file=/foo/invalid-config-1.json
unable to configure the Docker daemon with file /foo/invalid-config-1.json: the following directives don't match any configuration option: unknown-option
+ dockerd --validate --config-file=/foo/malformed-config.json
unable to configure the Docker daemon with file /foo/malformed-config.json: unexpected EOF
+ dockerd --validate --config-file=/foo/valid-config-1.json
INFO[2021-05-19T10:31:30.883917472Z] Config OK
```

Looking at the output:

- For a follow-up: I think we should ignore the "empty" file. It's probably ok to ignore an empty `daemon.json` (same as how we ignore empty json (`{}`)
- Looking at it now; using `logrus.Infof()` adds a lot of noise; I think we should just print `Config OK`. This also allows the output to be more easily used in scripts (check if the output is "Config OK"

I think we should also have a simple integration test for this. I see there's a `TestConfigDaemonLibtrustID` test in https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/integration/config/config_test.go#L382-L400

Some notes on that test though;

- it looks like it's "actually" in the wrong place (that test suite is testing `docker config` (swarm configs), so _perhaps_ should be moved to a new suite, but we can look at that later
- the "test utils" we have to start the daemon may be a bit complicated for this (they're mostly to setup a daemon and then interact with the running daemon, but not so much to test the daemon CLI); it would need a bit looking into (but happy to help if needed). This particular would probably have been easier to implement in the old `integration-cli` tests, but those were marked "deprecated", so we can't add new tests there. So instead, perhaps just using [`os/exec`](https://golang.org/pkg/os/exec/#Command) to start the binary and check stdout/stderr would work.